onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalsListFragment.java

```diff
@@ -1472,6 +1509,87 @@ private void setRoutesFilter(boolean[] checks) {
         setRoutesFilter(newFilter);
     }
 
+    private void setArrivalFilter(int selection) {
+        ArrivalFilter newFilter = ArrivalFilter.fromInt(selection);
+
+        boolean switchedFromOnlyDepartures =
+                mArrivalFilter == ArrivalFilter.ONLY_DEPARTURES &&
+                        newFilter != ArrivalFilter.ONLY_DEPARTURES;
+
+        mArrivalFilter = newFilter;
+
+        ObaContract.Stops.setArrivalFilter(getActivity(), mStopUri, ArrivalFilter.fromInt(selection));
+        refreshSituations(UIUtils.getAllSituations(getArrivalsLoader().getLastGoodResponse(), mRoutesFilter));
+        refreshLocal();
```

Comment:
I believe `refreshLocal();` should be moved to the end of this method as it triggers an adapter update.  We want to make sure that any UI content (e.g., "load more" text) is updated before triggering the reload.